### ISSUE TITLE
Inheriting from Page model to create another base page model for subset of pages produces incorrect migration fail on model delete

### ISSUE THREAD

**Issue by imomaliev** [wagtail/wagtail#7127]
When inheriting from Page model to create another base page model for subset of pages, and then deciding to delete or rename it fails with

### Issue Summary

Model structure
EventPage -> ContentPage -> Page

```python
  File "src/manage.py", line 22, in <module>
    main()
  File "src/manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/project/backend/.venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/project/backend/.venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 395, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/project/backend/.venv/lib/python3.8/site-packages/django/core/management/base.py", line 330, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/project/backend/.venv/lib/python3.8/site-packages/django/core/management/base.py", line 371, in execute
    output = self.handle(*args, **options)
  File "/project/backend/.venv/lib/python3.8/site-packages/django/core/management/base.py", line 85, in wrapped
    res = handle_func(*args, **kwargs)
  File "/project/backend/.venv/lib/python3.8/site-packages/django/core/management/commands/migrate.py", line 243, in handle
    post_migrate_state = executor.migrate(
  File "/project/backend/.venv/lib/python3.8/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/project/backend/.venv/lib/python3.8/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/project/backend/.venv/lib/python3.8/site-packages/django/db/migrations/executor.py", line 227, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/project/backend/.venv/lib/python3.8/site-packages/django/db/migrations/migration.py", line 114, in apply
    operation.state_forwards(self.app_label, project_state)
  File "/project/backend/.venv/lib/python3.8/site-packages/django/db/migrations/operations/fields.py", line 158, in state_forwards
    old_field = model_state.fields.pop(self.name)
KeyError: 'page_ptr'
```
This happens because during migration two entries for `RemoveField` are created instead of one

```python
        migrations.RemoveField(
            model_name='eventpage',
            name='contentpage_ptr',
        ),
        migrations.RemoveField(
            model_name='eventpage',
            name='page_ptr',
        ),
```

Here is what is causing this issue, it shouldn't have add this to migration. Deleting this fixes the problem

```python
        migrations.RemoveField(
            model_name='eventpage',
            name='page_ptr',
        ),
```

### Steps to Reproduce

TODO:
1. (for example) Start a new project with `wagtail start myproject`
2. Edit models.py as follows...
3. ...

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (yes / no)

### Technical details

* Python version: 3.8.8
* Django version: 3.1.8
* Wagtail version: 2.12.4
* Browser version: Server side problem

**Comment by gasman** [comment]
Thanks for the report @imomaliev! Since it's not really possible to investigate this further without steps to reproduce, I'll close this for now, but happy to reopen if you can provide those.

Since Wagtail doesn't override any of Django's migration logic here, it's possible that this is a Django bug - so if you do find a set of models that replicates the problem, it would be worth checking whether the same thing occurs when inheriting from a plain Django model rather than Page.

**Comment by imomaliev** [comment]
@gasman Thanks for your response. This issue is invalid. After more research I found out that our migration state broke somewhere. For some reason when we were trying to do data migration for `Page` grandchild

```
Page -> ContentPage -> EventPage
```

django thought there was no `ContentPage` in `__mro__`. So our state was wrong. I decided just to reset all migrations and create new initial ones

I think there is still a bug here but it is somewhere, because I know django is little finicky with updating bases for MTI. If I will be able to create reproducible  example I will share it here and create new issue, which as you suggested will probably be in django
